### PR TITLE
[feature]: zlib deflate concurrency limit

### PR DESCRIPTION
**Pull request by STRML**
Ref: #1202 

Built on top of [async-limiter](https://github.com/STRML/async-limiter) which is only 1kB of dependency-free ES5.

### CODE REVIEW

File: lib/PerMessageDeflate.js [hunk 1]
```diff
@@ -4,12 +4,21 @@ const safeBuffer = require('safe-buffer');
 const zlib = require('zlib');
 
 const bufferUtil = require('./BufferUtil');
+const Limiter = require('async-limiter');
```
Comment (lpinca) [inline lib/PerMessageDeflate.js]: Can you please move this to the above block?

File: lib/PerMessageDeflate.js [hunk 2]
```diff
@@ -249,14 +265,48 @@ class PerMessageDeflate {
...
+   * Decompress data. Concurrency limited by async-limiter.
    *
    * @param {Buffer} data Compressed data
    * @param {Boolean} fin Specifies whether or not this is the last fragment
    * @param {Function} callback Callback
    * @public
    */
   decompress (data, fin, callback) {
+    zlibLimiter.push((done) => {
+      this._decompress(data, fin, function (err, result) {
```
Comment (lpinca) [inline lib/PerMessageDeflate.js]: Mind to use an arrow function for consistency?

Comment (lpinca) [inline lib/PerMessageDeflate.js]: Mind to use arrow functions for consistency?

File: lib/PerMessageDeflate.js [hunk 3]
```diff
@@ -249,14 +265,48 @@ class PerMessageDeflate {
...
+   * Compress data. Concurrency limited by async-limiter.
+   *
+   * @param {Buffer} data Data to compress
+   * @param {Boolean} fin Specifies whether or not this is the last fragment
+   * @param {Function} callback Callback
+   * @public
+   */
+  compress (data, fin, callback) {
+    zlibLimiter.push((done) => {
+      this._compress(data, fin, function (err, result) {
```
Comment (lpinca) [inline lib/PerMessageDeflate.js]: Same here.

File: package.json [hunk 4]
```diff
@@ -26,6 +26,7 @@
     "lint": "eslint ."
   },
   "dependencies": {
+    "async-limiter": "^1.0.0",
```
Comment (lpinca) [inline package.json]: Can you please use the same versioning scheme?

**Comment by STRML** [comment]
Great - fixed nits.

**Comment by lpinca** [comment]
Thank you.

**Pull request merged**